src/learning/gmm.rs

```diff
@@ -70,17 +70,25 @@ pub struct GaussianMixtureModel {
 impl UnSupModel<Matrix<f64>, Matrix<f64>> for GaussianMixtureModel {
     /// Train the model using inputs.
     fn train(&mut self, inputs: &Matrix<f64>) -> LearningResult<()> {
+        use rulinalg::matrix::Axes;
+
         // Initialization:
         let k = self.comp_count;
 
-        // TODO: Compute sample covariance of the data.
-        // https://en.wikipedia.org/wiki/Sample_mean_and_covariance#Sample_covariance
-        let mut cov_vec = Vec::with_capacity(k);
-        for _ in 0..k {
-            cov_vec.push(Matrix::identity(inputs.cols()));
+        let means = inputs.mean(Axes::Col);
+
+        let mut cov_mat = Matrix::zeros(inputs.cols(), inputs.cols());
+        let reg_value = 1f64 / (inputs.rows() - 1) as f64;
```

Comment:
We will need to check here that we have `inputs.rows() > 1` and return an `Error` otherwise.

We may well do this check later when we call `variance` but we should catch it here before panicking on underflow (or producing `NaN` cov values when dividing by `0`).